core/crdt/stats.go

```diff
@@ -0,0 +1,306 @@
+/*
+
+ Warpnet - Decentralized Social Network
+ Copyright (C) 2025 Vadim Filin, https://github.com/Warp-net,
+ <github.com.mecdy@passmail.net>
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+ WarpNet is provided "as is" without warranty of any kind, either expressed or implied.
+ Use at your own risk. The maintainers shall not be liable for any damages or data loss
+ resulting from the use or misuse of this software.
+*/
+
+// Copyright 2025 Vadim Filin
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package crdt
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"sync"
+
+	crdt "github.com/ipfs/go-ds-crdt"
+	ds "github.com/ipfs/go-datastore"
+	dsquery "github.com/ipfs/go-datastore/query"
+	blockstore "github.com/ipfs/boxo/blockstore"
+	blockservice "github.com/ipfs/boxo/blockservice"
+	syncds "github.com/ipfs/go-datastore/sync"
+	offline "github.com/ipfs/boxo/exchange/offline"
+	dagservice "github.com/ipfs/boxo/ipld/merkledag"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
```

Comment:
You are not allowed to use external pubsub. There must be only one otherwise it panics. Use existing implementation: https://github.com/Warp-net/warpnet/blob/main/core/pubsub/gossip.go